### DISCUSSION TITLE
What is the best way of leading with errors in OMNeT++?

### DISCUSSION THREAD

**Discussion by carlosmn25** [omnetpp/omnetpp#1254] [Q&A]
We have a project based in OMNeT++ and we always have this doubt: how to deal with errors that imply finishing the simulation? From what we know, the "standard" what is to throw an exception such as "`cRuntimeException`", but this causes that our console is completely filled with "undisposed object" messages due to all the modules that are not being correctly deleted. This makes it quite hard (or directly impossible) to read the error message. Then, we thought that we could just capture the exception and call `endSimulation()` in the catch statement. That solves the problem of the console, but then, the simulation ends with exit code 0. This is the best solution that we were able to find until now, but it would be nice if we could stop simulations in an ordered way (properly deleting the objects) and with an exit code different from 0. We also tried calling `exit()` or `error()`, but we find the same problem of the "undisposed object" messages.

**Comment by rhornig** [comment]
Throw `cRuntimeError` and specify `print-undisposed = false` in the ini file to oppress the undisposed messages.
